textx/model.py

```diff
@@ -289,6 +297,62 @@ def get_model_from_str(self, model_str, file_name=None, debug=None,
                 pass
             return model
 
+        # Custom attr dunder methods used for user classes during loading
+        def _hasattr(obj, name):
+            try:
+                obj_attrs = obj.__class__._tx_obj_attrs
+                return name in obj_attrs[id(obj)]
+            except AttributeError:
+                return False
+
+        def _getattr(obj, name):
+            try:
+                return obj._tx_obj_attrs[id(obj)][name]
+            except KeyError:
+                raise AttributeError
+
+        def _setattr(obj, name, value):
+            obj._tx_obj_attrs[id(obj)][name] = value
+
+        def _delattr(obj, name):
+            obj._tx_obj_attrs[id(obj)].pop(name)
+
+        def _replace_user_attr_methods(self):
+            """
+            Replace has/get/set/del(attr) methods on user classes
+            to support postponing of user obj initialization.
+            """
+            for user_class in self.metamodel.user_classes.values():
+                if not hasattr(user_class, '_tx_instrumented'):
+                    for a_name in ('has', 'get', 'set', 'del'):
+                        real_name = '__{}attr__'.format(a_name)
+                        cached_name = '_tx_old_{}attr'.format(a_name)
+                        setattr(user_class, cached_name,
+                                getattr(user_class, real_name, None))
+                        setattr(user_class, real_name,
+                                getattr(self.__class__,
+                                        '_{}attr'.format(a_name)))
+                    setattr(user_class, '_tx_instrumented', True)
```

Comment:
We should make this a count which we increase if this method is called for the same class again:
```python
if not hasattr(user_class, '_tx_instrumented'):
    ...
else:
    user_class._tx_instrumented += 1